### PR TITLE
Allow arrays to be typed as ICollection with DontInline

### PR DESCRIPTION
**Pull request by lbargaoanu**
Technically this is a fix for #1973, but I'm not sure it would make sense to the regular user. It's a bug regardless.

### CODE REVIEW

**Comment by jbogard** [comment]
Hmmmm, should we have a better name for that method? It's really talking about the implementation vs. what it's doing. Maybe something with "Runtime" in the name, since it's switching from config-time type checking to runtime type checking?

**Comment by lbargaoanu** [comment]
Yes, I was thinking MapAtRuntime, but I'm not sure if it fits the original use case.

**Comment by jbogard** [comment]
Ah right the original use case was "the expression tree is large enough to crash the process".

**Comment by lbargaoanu** [comment]
Rebased.

File: src/AutoMapper/Configuration/MemberConfigurationExpression.cs [hunk 1]
```diff
@@ -20,7 +20,7 @@ public MemberConfigurationExpression(MemberInfo destinationMember, Type sourceTy
 
         public MemberInfo DestinationMember => _destinationMember;
 
-        public void DoNotInline()
+        public void MapAtRuntime()
         {
             PropertyMapActions.Add(pm => pm.Inline = false);
```
Comment (jbogard) [inline src/AutoMapper/Configuration/MemberConfigurationExpression.cs]: Should we rename this property too?

Comment (lbargaoanu) [inline src/AutoMapper/Configuration/MemberConfigurationExpression.cs]: I'd say no. I think for us inline makes more sense. MapAtRuntime is a little vague :)

**Pull request merged**